### DISCUSSION TITLE
Rename and copy moved out of post render module

### DISCUSSION THREAD

**Discussion by JamesBotterill** [inlife/nexrender#833] [Ideas]
As the default export in AE2023 is mp4, it seems like one of the main reasons to have the nexrender-action-copy as a separate module has been removed. 

That being you'd normally need to run script after the original avi/mov has been encoded to mp4. 

Wouldn't it be cleaner to specify an output dir and filename as part of the template settings, it would still be prudent to retain this module in cases where users are encoding away from the default job.output file. 

The two options I would have are as follows:

```
outputFilename: string
outputDirectory: string
```
These could be processed at the end of the postrender actions phase pre-clean up.

**Comment by inlife** [comment]
I agree that having something like that would be great, but it seems to me that adding those two parameters will become confusing, especially if we already have `job.output`, which does the same thing.

You can always provide: `job.output: "<Absolute path>"` as the instruction and it will work as expected.

And for the keeping actions for now, I think is a good idea, since the adoption rate of the newer AE versions is not always 100%, my assumption some people are still using even AE 2016 sometimes. :D

**Reply by JamesBotterill** [reply]
Yes I see and that would make sense. 

in my scenario it would purely be replicating the `path.join(job.workpath, job.resultname)` in the setup. 

so hypothetically if I set the filename (including extension) and path as absolutes with the job.output it will always output as that to that location.

I'll test this tomorrow

**Reply by JamesBotterill** [reply]
If you do this without adding the file extension then it works as you described.

I'll add a pr do document this method as it's far more efficient.
